main.go

```diff
@@ -506,7 +506,158 @@ func downloadByUrl(c echo.Context) error {
+	if err != nil {
+		log.Printf("Error querying BD (filter or counter):%q", err)
+		return c.JSON(http.StatusInternalServerError, err.Error())
+	}
+
+	remunerations, numRows, err := getSearchResults(results, filter.Category)
+	if err != nil {
+		log.Printf("Error getting search results: %q", err)
+		return c.JSON(http.StatusInternalServerError, err.Error())
+	}
+	// Devemos retornar um array vazio quando a pesquisa não retornar dados.
+
+	response := models.SearchResponse{
+		DownloadAvailable:  numRows > 0 && numRows <= conf.DownloadLimit,
+		NumRowsIfAvailable: numRows,
+		DownloadLimit:      conf.DownloadLimit,
+		SearchLimit:        conf.SearchLimit,
+		Results:            remunerations, // retornando os SearchLimit primeiros elementos.
+	}
+	return c.JSON(http.StatusOK, response)
+}
+
+func lowCostDownload(c echo.Context) error {
+	var years string
+	var months string
+	var agencies string
+	var categories string
+	var types string
+	//Pegando os query params
+	years = c.QueryParam("anos")
+	months = c.QueryParam("meses")
+	agencies = c.QueryParam("orgaos")
+	categories = c.QueryParam("categorias")
+	types = c.QueryParam("tipos")
+
+	//Criando os filtros a partir dos query params e validando eles
+	filter, err := models.NewFilter(years, months, agencies, categories, types)
+	if err != nil {
+		return c.JSON(http.StatusBadRequest, err.Error())
+	}
+
+	results, err := postgresDb.lowCostFilter(lowCostRemunerationQuery(filter), lowCostArguments(filter))
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, err.Error())
+	}
+
+	searchResults, _, err := getSearchResults(results, filter.Category)
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, err.Error())
+	}
+
+	c.Response().Header().Set("Content-Disposition", "attachment; filename=dadosjusbr-remuneracoes.csv")
+	c.Response().Header().Set("Content-Type", c.Response().Header().Get("Content-Type"))
+	err = gocsv.Marshal(searchResults[:conf.DownloadLimit], c.Response().Writer)
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, fmt.Errorf("erro tentando fazer download do csv: %q", err))
+	}
+	return nil
+}
+
+func getSearchResults(results []models.SearchDetails, category string) ([]models.SearchResult, int, error) {
+	searchResults := []models.SearchResult{}
+
+	numRows := 0
+	if len(results) == 0 {
+		return searchResults, numRows, nil
+	} else {
+		// A razão para essa ordenação é que quando o usuário escolhe diversos órgãos
+		// provavelmente ele prefere ver dados de todos eles. Dessa forma, aumentamos
+		// as chances do preview limitado retornar dados de diversos órgãos.
+		sort.SliceStable(results, func(i, j int) bool {
+			return results[i].Ano < results[j].Ano || results[i].Mes < results[j].Mes
+		})
+		// forma de evitar a descompactação de arquivos que estão
+		// fora do limite e, por isso, não serão retornados.
+		mustUnzip := true
+		for _, r := range results {
+			if mustUnzip {
+				remunerations, err := getRemunerationsFromZip(r.ZipUrl)
```

Comment:
Aqui você deve utilizar uma operação em batch, usando a SDK do AWS. Vai ser muito mais eficiente e barato que realizar dezenas de requisições HTTP.

https://aws.amazon.com/pt/blogs/developer/aws-sdk-for-go-batch-operations-with-amazon-s3/